Fix "Compound selectors may longer be extended." error when compiling

mattfelten opened the pull request:
Fixes #551 

More recently updated SCSS compilers have removed the ability to extend compound selectors, so we can no longer extend psuedoelements like `.test::before` and `.test::after`. We have to extend a `%placeholder` instead.

Refactored code to use placeholders and extending those.

joneff commented:
Thank you for the pull request. I will review it ASAP.

joneff commented:
@mattfelten, can you reword your commit message like mentioned in our guidelines -- https://github.com/telerik/kendo-themes/wiki/Commit-Message-Guidelines? Namely, `fix: Compound ....`.

The code itself is logically refactored. I am not sure about `%k-i-loading-psuedo` naming, though. `%k-i-loading-before-after` perhaps? Alternatively, we can simply combine selectors and avoid the extending altogether. How does it sound to you?

mattfelten commented:
Updated! Changed the commit message (needed to `git push --force`) and removed the extends completely. You're right. We didn't really even need to do that.
